### PR TITLE
gnrc/6lo: Set more data flag on all but last fragment

### PR DESCRIPTION
**Pull request by gebart**
### Contribution description

Use the flag introduced in #9469 to signal to the link layer that more data will follow this packet on outgoing fragments, except for the final frag of the datagram.

### Issues/PRs references

Depends on #9469 
Does not do much difference without also merging #9470

### CODE REVIEW

**Comment by gebart** [comment]
With ContikiMAC (unpublished) this improves network latency significantly for fragmented unicast packets.

**Comment by PeterKietzmann** [comment]
The change seems reasonable although I'd like to get @miri64's feedback here. Furthermore, I wanted to reference [this mail](https://lists.riot-os.org/pipermail/devel/2018-June/005795.html) as a reminder to ongoing fragmentation efforts.

**Comment by miri64** [comment]
The reassembly/fragmentation work did not start yet (see lacking replies to the mail you referenced). However, I'm currently refactoring 6Lo with the goal to get Fragmentation and IPHC more separated (see https://github.com/RIOT-OS/RIOT/projects/18 for progress). I don't think however, that that much collides with that effort and should be easily rebasable when it comes to conflicts.

**Comment by PeterKietzmann** [comment]
>  I don't think however, that that much collides with that effort and should be easily rebasable when it comes to conflicts.

Fine,  let's wait for #9469 and #9470 then. I can do testing afterwards and we go with this PR as well (if everything works as expected).

**Comment by miri64** [comment]
Can you please rebase.

**Comment by gebart** [comment]
rebased

File: sys/net/gnrc/network_layer/sixlowpan/frag/gnrc_sixlowpan_frag.c [hunk 1]
```diff
@@ -243,7 +259,7 @@ void gnrc_sixlowpan_frag_send(gnrc_pktsnip_t *pkt, void *ctx, unsigned page)
     }
 #endif
 
-    /* Check weater to send the first or an Nth fragment */
+    /* Check whether to send the first or an Nth fragment */
```
Comment (miri64) [inline sys/net/gnrc/network_layer/sixlowpan/frag/gnrc_sixlowpan_frag.c]: Please put this spelling fix in a separate commit.

Comment (gebart) [reply sys/net/gnrc/network_layer/sixlowpan/frag/gnrc_sixlowpan_frag.c]: done

**Comment by miri64** [comment]
> 6lo: Speling correction in comment

Spe*ll*ing :laughing:

**Comment by miri64** [comment]
Okay :grinning:
